test/Microsoft.ML.Tests/Transformers/ValueMappingTests.cs

```diff
@@ -0,0 +1,157 @@
+                new int[] {2, 3, 4 },
+                new int[] {100, 200 },
+                new int[] {400, 500, 600, 700 }};
+
+            var estimator = new ValueMappingEstimator<ReadOnlyMemory<char>, int>(Env, keys, values, new[] { ("A", "D"), ("B", "E"), ("C", "F") });
+            var t = estimator.Fit(dataView);
+
+            var result = t.Transform(dataView);
+            var cursor = result.GetRowCursor((col) => true);
+            var getterD = cursor.GetGetter<VBuffer<int>>(3);
+            var getterE = cursor.GetGetter<VBuffer<int>>(4);
+            var getterF = cursor.GetGetter<VBuffer<int>>(5);
+            cursor.MoveNext();
+
+            var valuesArray = values.ToArray();
+            VBuffer<int> dValue = default;
+            getterD(ref dValue);
+            Assert.Equal(values[1].Length, dValue.Length);
+            VBuffer<int> eValue = default;
+            getterE(ref eValue);
+            Assert.Equal(values[2].Length, eValue.Length);
+            VBuffer<int> fValue = default;
+            getterF(ref fValue);
+            Assert.Equal(values[0].Length, fValue.Length);
+        }
+
+        [Fact]
+        public void ValueMappingMissingKey()
+        {
+            var data = new[] { new TestClass() { A = "barTest", B = "test", C = "foo" } };
+            var dataView = ComponentCreation.CreateDataView(Env, data);
+
+            IEnumerable<ReadOnlyMemory<char>> keys = new List<ReadOnlyMemory<char>>() { "foo".AsMemory(), "bar".AsMemory(), "test".AsMemory(), "wahoo".AsMemory() };
+            IEnumerable<int> values = new List<int>() { 1, 2, 3, 4 };
+
+            var estimator = new ValueMappingEstimator<ReadOnlyMemory<char>, int>(Env, keys, values, new[] { ("A", "D"), ("B", "E"), ("C", "F") });
+            var t = estimator.Fit(dataView);
+
+            var result = t.Transform(dataView);
+            var cursor = result.GetRowCursor((col) => true);
+            var getterD = cursor.GetGetter<int>(3);
+            var getterE = cursor.GetGetter<int>(4);
+            var getterF = cursor.GetGetter<int>(5);
+            cursor.MoveNext();
+
+            int dValue = 1;
+            getterD(ref dValue);
+            Assert.Equal(0, dValue);
+            int eValue = 0;
+            getterE(ref eValue);
+            Assert.Equal(3, eValue);
+            int fValue = 0;
+            getterF(ref fValue);
+            Assert.Equal(1, fValue);
+        }
+
+
+        [Fact]
+        public void ValueMappingWorkout()
+        {
+            var data = new[] { new TestClass() { A = "bar", B = "test", C = "foo" } };
+            var dataView = ComponentCreation.CreateDataView(Env, data);
+            var badData = new[] { new TestWrong() { A = "bar", B = 1.2f } };
+            var badDataView = ComponentCreation.CreateDataView(Env, badData);
+
+            IEnumerable<ReadOnlyMemory<char>> keys = new List<ReadOnlyMemory<char>>() { "foo".AsMemory(), "bar".AsMemory(), "test".AsMemory(), "wahoo".AsMemory() };
```

Comment:
>ReadOnlyMemory<char> [](start = 24, length = 20)

What would happen if I pass string?
Aren't we cast string to readonlymemory during dataview construction?

Reply:
Ping

---
In reply to: [237660977](https://github.com/dotnet/machinelearning/pull/1710#discussion_r237660977) [](ancestors = 237660977)

Reply:
I would prefer that since this would contain the Readonly memory, let me look into what that would take. That could be something to add in another review.